tools/build.py

```diff
@@ -0,0 +1,653 @@
+                        help="Specify the target Android Application Binary Interface (ABI)")
+    parser.add_argument("--android_api", type=int, default=27, help="Android API Level, e.g. 21")
+    parser.add_argument("--android_home", type=Path, default=os.environ.get("ANDROID_HOME"),
+                        help="Path to the Android SDK.")
+    parser.add_argument("--android_ndk_path", type=Path, default=os.environ.get("ANDROID_NDK_HOME"),
+                        help="Path to the Android NDK. Typically `<Android SDK>/ndk/<ndk_version>")
+
+    # macOS/iOS options
+    parser.add_argument("--build_apple_framework", action="store_true",
+                        help="Build a macOS/iOS framework for the ONNXRuntime.")
+    parser.add_argument("--ios", action="store_true", help="build for iOS")
+    parser.add_argument("--ios_sysroot", default="",
+                        help="Specify the name of the platform SDK to be used. e.g. iphoneos, iphonesimulator")
+    parser.add_argument("--ios_toolchain_file", default=f"{REPO_DIR}/cmake/ortext_ios.toolchain.cmake", type=Path,
+                        help="Path to ios toolchain file. Default is <repo>/cmake/ortext_ios.toolchain.cmake")
+    parser.add_argument("--xcode_code_signing_team_id", default="",
+                        help="The development team ID used for code signing in Xcode")
+    parser.add_argument("--xcode_code_signing_identity", default="",
+                        help="The development identity used for code signing in Xcode")
+    parser.add_argument("--osx_arch", default="arm64" if platform.machine() == "arm64" else "x86_64",
+                        choices=["arm64", "arm64e", "x86_64"],
+                        help="Specify the Target specific architectures for macOS and iOS. "
+                             "This is only supported on macOS")
+    parser.add_argument("--apple_deploy_target", type=str,
+                        help="Specify the minimum version of the target platform (e.g. macOS or iOS). "
+                             "This is only supported on macOS")
+
+    # WebAssembly options
+    parser.add_argument("--wasm", action="store_true", help="Build for WebAssembly")
+    parser.add_argument("--emsdk_path", type=Path,
+                        help="Specify path to emscripten SDK. Setup manually with: "                         
+                             "  git clone https://github.com/emscripten-core/emsdk")
+    parser.add_argument("--emsdk_version", default="3.1.26", help="Specify version of emsdk")
+
+    # x86 args
+    parser.add_argument("--x86", action="store_true",
+                        help="[cross-compiling] Create Windows x86 makefiles. Requires --update and no existing cache "
+                             "CMake setup. Delete CMakeCache.txt if needed")
+
+    # Arguments needed by CI
+    parser.add_argument("--cmake_path", default="cmake", type=Path, help="Path to the CMake program.")
+    parser.add_argument("--ctest_path", default="ctest",
```

Comment:
nit: maybe just make it consistent?
`parser.add_argument("--ctest_path", default="ctest", type=Path, help="Path to the CTest program.")`